### PR TITLE
Smart contract new info

### PR DESCRIPTION
**Pull request by mblockelet**
This pull request adds new information provided by Completium, such as :
* internal transactions
* current balance after each call
* level, now, ...

and improves some display elements related to smart contracts.

### CODE REVIEW

File: frontend/task/libs/smart_contract/SmartContractView.tsx [hunk 1]
```diff
@@ -35,12 +35,29 @@ export function SmartContractView() {
...
+                if (log.source && (undefined !== balances[log.source] || !isContract(log.source))) {
+                    // if undefined, it's not a contract and starts with 10000 tez
+                    balances[log.source] = (balances[log.source] ?? 10000) - Number(log.amount);
+                }
+                if (log.destination && (undefined !== balances[log.destination] || !isContract(log.destination))) {
+                    // if undefined, it's not a contract and starts with 10000 tez
+                    balances[log.destination] = (balances[log.destination] ?? 10000) + Number(log.amount);
+                }
+            }
+            log.balances = Object.assign({}, balances);
```
Comment (SebastienTainon) [inline frontend/task/libs/smart_contract/SmartContractView.tsx:53]:
```suggestion
            log.balances = {...balances};
```

You could use the ES6 spread syntax, shorter and easier to read ;)

File: frontend/task/libs/smart_contract/smart_contract_lib.ts [hunk 2]
```diff
@@ -141,7 +151,12 @@ export class SmartContractLib extends QuickAlgoLibrary {
             }
 
             return new LibraryTestResult(
-                output.error?.message
+                output.error?.message,
+                'task-submission-test-result-smart-contract',
```
Comment (SebastienTainon) [inline frontend/task/libs/smart_contract/smart_contract_lib.ts:155]: What is this part for? I don't see where it's used

**Review by SebastienTainon — APPROVED** [review]

File: frontend/task/libs/smart_contract/SmartContractView.tsx (hunk 1 above)
Comment (mblockelet) [reply frontend/task/libs/smart_contract/SmartContractView.tsx]: 👍

File: frontend/task/libs/smart_contract/smart_contract_lib.ts (hunk 2 above)
Comment (mblockelet) [reply frontend/task/libs/smart_contract/smart_contract_lib.ts]: :+1:

**Pull request merged**